schema/tests/schema_json.rs

```diff
@@ -0,0 +1,239 @@
+//! This test checks how the json-serialized schema looks like.
+
+#![allow(dead_code)]
+#![allow(unused_tuple_struct_fields)]
+
+use iroha_schema::IntoSchema;
+use serde_json::json;
+
+macro_rules! check_schema_impl {
+    ($ty:ident, $type_name:ident, $json:tt) => {{
+        assert_eq!(
+            $ty::type_name(),
+            stringify!($type_name),
+            "Type name of {} is not equal to the expected one",
+            stringify!($ty)
+        );
+        let __schema = serde_json::value::to_value(&$ty::schema())
+            .expect("Failed to serialize schema to JSON");
+        // println!("{}: {}", stringify!($ty), __schema);
+        assert_eq!(
+            __schema.get(&$ty::type_name()).unwrap().clone(),
+            json!($json),
+            "Schema of {} is not equal to the expected one",
+            stringify!($ty)
+        );
+    }};
+}
+
+/// It expects to have three parameters: a type definition item, an expected schema type name and a JSON schema.
+///
+/// The json is passed to the `serde_json::json!` macro, so it can be a string, an array or an object.
+///
+/// Only the schema of the type itself is checked, not the schema of its fields.
+///
+/// NOTE: this macro doesn't support generics.
+macro_rules! check_schema {
+    ($(#[$($meta:tt)*])* struct $ty:ident, $type_name:ident, $json:tt) => {{
+        #[derive(IntoSchema)]
+        $(#[$($meta)*])*
+        struct $ty;
+        check_schema_impl!($ty, $type_name, $json);
```

Comment:
Better use [this technique](https://stackoverflow.com/questions/54406388/what-does-an-symbol-mean-in-a-rust-declarative-macro)

Reply:
So, instead of defining an `_impl` macro, I should create a matcher with `@impl` containing the same stuff? Sounds neat